addons/mail/static/src/models/thread/thread.js

```diff
@@ -1602,8 +1615,20 @@ function factory(dependencies) {
                 'serverFoldState',
             ],
         }),
+        /**
+         * All messages ordered like they are displayed. This field does not
+         * contain transient messages which are not "real" records.
+         */
         orderedMessages: many2many('mail.message', {
             compute: '_computeOrderedMessages',
+            dependencies: ['orderedMessagesIncludingTransient'],
+        }),
+        /**
+         * All messages ordered like they are displayed. This field can contain
+         * transient messages which are not "real" records.
+         */
+        orderedMessagesIncludingTransient: many2many('mail.message', {
```

Comment:
You'll have to use this one when rendering messages in message list if we want transient messages to be shown.

Reply:
In fact message list is using `orderedMessages` from threadCache which contains transients

Reply:
Ok but this new field is not used anywhere then?

Reply:
Should thread_preview consider transient messages to determine last message ?
The same question for thread last message.

If they don't, then no, this field is not used and should not.
If they do, then no, this field is not used but should be.

Reply:
Thread preview should show the last real message as last message, not the transient message.

Reply:
And for the "thread last message" what do you mean?

Reply:
There is a computed field on thread model "lastMessage" which is based on orderedMessages which is also used in thread preview, thread needaction preview and notification list

Reply:
For all of those that shown on the messaging menu it's good to ignore transient, so I guess you can remove this new field here and just ensure the original one is excluding transient.